### PR TITLE
Display project path in error message

### PR DESCRIPTION
**Pull request by MDrakos**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-1254" title="DX-1254" target="_blank">StoryDX-1254</a>  When applicable, display the org/project and path of a project not found
  </td></table>
  <br />

### CODE REVIEW

File: internal/runbits/findproject/project.go [hunk 1]
```diff
@@ -119,7 +119,16 @@ func FromNamespaceLocal(ns *project.Namespaced, cfg projectfile.ConfigGetter, pr
...
 
-		return project.FromPath(path)
+		proj, err := project.FromPath(path)
+		if err != nil {
+			if errs.Matches(err, &projectfile.ErrorNoProject{}) {
+				return nil, &LocalProjectDoesNotExist{
+					locale.WrapInputError(err, "err_local_project_not_checked_out", "", ns.Project),
+				}
+			}
+			return nil, locale.WrapError(err, "err_project_frompath", "Could not load project from path: {{.V0}}", path)
```
Comment (mitchell-as) [inline internal/runbits/findproject/project.go:129]: This locale key already exists, so please reconcile the two.

**Review by mitchell-as — CHANGES_REQUESTED** [review]
This looks good other than one small issue. Please flag Nathan for final review though.

**Review by mitchell-as — DISMISSED** [review]

File: internal/runbits/findproject/project.go [hunk 2]
```diff
@@ -119,7 +119,16 @@ func FromNamespaceLocal(ns *project.Namespaced, cfg projectfile.ConfigGetter, pr
...
 
-		return project.FromPath(path)
+		proj, err := project.FromPath(path)
+		if err != nil {
+			if errs.Matches(err, &projectfile.ErrorNoProject{}) {
+				return nil, &LocalProjectDoesNotExist{
+					locale.WrapInputError(err, "err_local_project_not_checked_out", "", ns.Project),
+				}
+			}
+			return nil, locale.WrapError(err, "err_findproject_frompath", "Could not load project from path: {{.V0}}", path)
```
Comment (Naatan) [inline internal/runbits/findproject/project.go:129]: We should mention the namespace here as well imo.

File: internal/runbits/findproject/project.go [hunk 3]
```diff
@@ -68,7 +68,7 @@ func FromNamespaceLocal(ns *project.Namespaced, cfg projectfile.ConfigGetter, pr
 
 	matchingProjects := make(map[string][]string)
 	matchingNamespaces := make([]string, 0)
-	for namespace, paths := range projectfile.GetProjectMapping(cfg) {
+	for namespace, paths := range projectfile.GetStaleProjectMapping(cfg) {
```
Comment (Naatan) [inline internal/runbits/findproject/project.go:71]: This doesn't feel right. We might be prompting them to use a project that no longer exists.. that feels very strange.

I don't think we need to be doing this at all. If a user deleted a project then we can't find it. Offering the original location feels like it's just going to cause confusion and buggy code.

In terms of the story I would consider this scenario as "not applicable".

**Review by Naatan — CHANGES_REQUESTED** [review]

File: internal/runbits/findproject/project.go (hunk 3 above)
Comment (MDrakos) [reply internal/runbits/findproject/project.go]: Ahh right, I hadn't thought about the prompting. Update this function to store the stale project listing and only use it when we can't find a project locally.

File: pkg/project/namespace.go [hunk 4]
```diff
@@ -93,6 +94,15 @@ func (ns *Namespaced) Validate() error {
 	return nil
 }
 
+func (ns *Namespaced) Equal(other *Namespaced) bool {
+	if ns == nil || other == nil {
+		return false
+	}
+
+	return !ns.AllowOmitOwner && strings.EqualFold(strings.ToLower(other.String()), strings.ToLower(ns.String())) ||
```
Comment (Naatan) [inline pkg/project/namespace.go:102]: This has multiple issues:

* You're not checking whether AllowOmitOwner matches. And `Equal()` makes no implications about this behavior.
* Either namespace can include commit ID's. The current behavior would want the commit ID's  to match also. 

I suggest naming this something a bit more explicit. eg. `OwnerIsEqual()` and `ProjectIsEqual()`.

**Review by Naatan — CHANGES_REQUESTED** [review]

File: pkg/project/namespace.go (hunk 4 above)
Comment (MDrakos) [reply pkg/project/namespace.go]: Rather than having a convenience function I've moved the checks back to where they were initially.

**Pull request merged**